tests/test_compressors/quantized_compressors/test_packed_asym_decompression.py

```diff
@@ -0,0 +1,224 @@
+
+
+class SimpleModel(Module):
+    """Simple model for testing"""
+    def __init__(self, input_dim=512, hidden_dim=256, output_dim=128):
+        super().__init__()
+        self.layer1 = Linear(input_dim, hidden_dim, bias=False)
+        self.layer2 = Linear(hidden_dim, output_dim, bias=False)
+    
+    def forward(self, x):
+        x = self.layer1(x)
+        x = torch.relu(x)
+        x = self.layer2(x)
+        return x
+
+
+def create_asymmetric_quant_config(
+    num_bits=4,
+    strategy=QuantizationStrategy.GROUP,
+    group_size=128
+) -> QuantizationConfig:
+    """Create an asymmetric quantization config"""
+    config_groups = {
+        "group_1": QuantizationScheme(
+            targets=["Linear"],
+            weights=QuantizationArgs(
+                num_bits=num_bits,
+                strategy=strategy.value,
+                group_size=group_size if strategy == QuantizationStrategy.GROUP else None,
+                symmetric=False,
+            ),
+        ),
+    }
+    return QuantizationConfig(config_groups=config_groups)
+
+
+@pytest.mark.parametrize(
+    "strategy,group_size",
+    [
+        (QuantizationStrategy.GROUP, 128),
+        (QuantizationStrategy.CHANNEL, None),
+    ],
+)
+def test_end_to_end_asymmetric_quantization(
+    strategy,
+    group_size,
+    mock_per_group_calibration,
+    mock_per_channel_calibration,
+):
+    """
+    Test end-to-end workflow: quantize -> compress -> save -> load -> decompress -> use
+    """
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        tmp_path = Path(tmp_dir)
+        
+        model = SimpleModel()
+        original_weights = {
+            "layer1": model.layer1.weight.detach().clone(),
+            "layer2": model.layer2.weight.detach().clone(),
+        }
+        
+        quant_config = create_asymmetric_quant_config(
+            num_bits=4,
+            strategy=strategy,
+            group_size=group_size
+        )
+        # Set pack-quantized format for ModelCompressor usage
+        quant_config.format = CompressionFormat.pack_quantized.value
+        apply_quantization_config(model, quant_config)
+
+        if strategy == QuantizationStrategy.GROUP:
+            mock_per_group_calibration(model.layer1, "weight", model.layer1.weight, group_size)
+            mock_per_group_calibration(model.layer2, "weight", model.layer2.weight, group_size)
+        else:
+            mock_per_channel_calibration(model.layer1, "weight", model.layer1.weight)
+            mock_per_channel_calibration(model.layer2, "weight", model.layer2.weight)
+        
+        
+        
+        compressor = PackedQuantizationCompressor(config=quant_config)
+        quantized_modules_to_scheme = {
+            "layer1": quant_config.config_groups["group_1"],
+            "layer2": quant_config.config_groups["group_1"],
+        }
+        
+        state_dict = model.state_dict()
+        compressed_state_dict = compressor.compress(
```

Comment:
It may be simpler to use compress_model / decompress_model, both of which will work on the model in memory

compress / decompress are primarily meant for disk